ibis/backends/pandas/aggcontext.py

```diff
@@ -620,13 +621,23 @@ def agg(
                 mask = ~(window_sizes.isna())
                 window_upper_indices = pd.Series(range(len(window_sizes))) + 1
                 window_lower_indices = window_upper_indices - window_sizes
+                # The result Series of udf may need to be trimmed by
+                # timecontext. In order to do so, 'time' must be added
+                # as an index to the Series, if present. Here We extract
+                # time column from the parent Dataframe `frame`.
+                if TIME_COL in frame or TIME_COL in frame.index.names:
```

Comment:
When would TIME_COL be in frame.index.names? Shouldn't time always be a column?

Reply:
There are cases when `TIME_COL` used as the index for df, aren't there?

Reply:
I see your point of removing this, ibis should not set TIME_COL to be index